braindecode/models/eegitnet.py

```diff
@@ -0,0 +1,234 @@
+        classification)
+    n_in_chans: int
+        number of input EEG channels
+    input_window_samples : int
+        Number of time samples.
+    drop_prob: float
+        Dropout probability.
+
+    References
+    ----------
+    .. [Salami2022] A. Salami, J. Andreu-Perez and H. Gillmeister, "EEG-ITNet: An Explainable
+    Inception Temporal Convolutional Network for motor imagery classification," in IEEE Access,
+    doi: 10.1109/ACCESS.2022.3161489.
+    
+    Notes
+    -----
+    This implementation is not guaranteed to be correct, has not been checked
+    by original authors, only reimplemented from the paper based on author implementation.
+    """
+
+    def __init__(self, n_classes, in_channels, input_window_samples,
+                 drop_prob=0.4):
+        super().__init__()
+
+        
+        # ======== handling EEG input ========================
+        def _permute(x):
+            """
+            Permutes data:
+
+            from dim:
+            batch, chans, time, 1
+
+            to dim:
+            batch, 1, chans, time
+
+            """
+            return x.permute([0, 3, 1, 2])
+
+        self.prep_input = nn.Sequential(
+            Ensure4d(),
+            Expression(_permute))
+        # ======== Inception branches ========================
+        self.inception_branches=_get_inception_branch(in_ch=1, n_blocks=3, in_channels=in_channels)
+        self.pool1 = nn.Sequential(
+            *[nn.AvgPool2d(kernel_size=(1, 4)), nn.Dropout(drop_prob)]
+        )
+
+        # =========== TC blocks =====================
+        self.tc_block1=_TCBlock(in_ch=14, kernel_sz=4, dialation=1, padding=3)
+        # ================================
+        self.tc_block2=_TCBlock(in_ch=14, kernel_sz=4, dialation=2, padding=6)
+        # ================================
+        self.tc_block3=_TCBlock(in_ch=14, kernel_sz=4, dialation=4, padding=12)
+        # ================================
+        self.tc_block4=_TCBlock(in_ch=14, kernel_sz=4, dialation=8, padding=24)
+
+        # ============= dimensionality reduction ===================
+        self.block_reduce = nn.Sequential(
+            *[
+                nn.Conv2d(14, 28, kernel_size=(1, 1)),
+                nn.BatchNorm2d(28),
+                nn.ELU(),
+                nn.AvgPool2d((1, 4)),
+                nn.Dropout(drop_prob),
+            ]
+        )
+        # ============== Classifier ==================
+        self.classifier = nn.Linear(
+            int(int(input_window_samples / 4) / 4) * 28, n_classes
+        )
+        self.m = nn.Softmax(dim=1)
+        # ================================
+
+    def forward(self, x):
+        x = self.prep_input(x)
+        # ================================
+        x=self.inception_branches(x)
+        x = self.pool1(x)
+        x1 = x
+        # ============= TC blocks with residual connection ===================
+        x=self.tc_block1(x)
+        x = x + x1
+        x1 = x
```

Comment:
if we implement this inside `_TCBlock` forward we can omit residual connection here, it will be done inside `_TCBlock` forward 😄

Reply:
added in the `_TCBlock`